gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -895,7 +896,8 @@ private long computeCompletenessWatermark(String table, SortedSet<ZonedDateTime>
       while (iterator.hasNext()) {
         ZonedDateTime timestampDT = iterator.next();
         if (timestampDT.isAfter(prevWatermarkDT)
-            && TimeIterator.durationBetween(prevWatermarkDT, now, granularity) > 1) {
+            && TimeIterator.durationBetween(prevWatermarkDT, now, granularity) > 1
+            && TimeIterator.durationBetween(timestampDT, now, granularity) > 0) {
```

Comment:
To make the code readable and for future fixes: it maybe worthwhile to wrap this condition into a method called "isCurrentTimeAfterCompletenessWindow" or something more intuitive?